x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/jdbc/JdbcPreparedStatementTests.java

```diff
@@ -550,7 +550,8 @@ public void testThrownExceptionsWhenSettingByteArrayValues() throws SQLException
     }
 
     private long randomMillisSinceEpoch() {
-        return randomLongBetween(0, System.currentTimeMillis());
+        // random between Jan 1st, 1970 and Jan 1st, 2050
```

Comment:
Mostly just curious: Why this end bound? Also, should we also be testing epochMillis that are before 1970-01-01 as well since they are perfectly valid?

Reply:
I saw this one before in ResultSetBaseTestCase. Maybe it's time to move it to ESTestCase along with other `random*(....)` methods. I agree with @colings86  - it would be nice to get negative timestamps from time to time for testing. But I am +1 on constant upper bound since it will make the failure reproducible in (possibly very rare) case it somehow fails because of a particular timestamp.

Reply:
@colings86 @imotov since the suggestion is to move this to ESTestCase base class (and be part of the test infra in ES), any good ideas for the interval start date?

Reply:
I'm not sure why `randomLong()` is not good enough here? All long values are valid timestamps since epoch so should be not just use that?

Reply:
:+1:  That sounds like a great idea. As long as we don't pass that to H2 we should be fine.